src/status_im2/contexts/chat/lightbox/utils.cljs

```diff
@@ -15,24 +15,37 @@
     [status-im2.contexts.chat.lightbox.constants :as constants]
     [utils.worklets.lightbox :as worklet]))
 
+(defn clear-timers
+  [timers]
+  (js/clearTimeout (:mount-0 @timers))
+  (js/clearTimeout (:mount-1 @timers))
+  (js/clearTimeout (:hide-0 @timers))
+  (js/clearTimeout (:hide-1 @timers))
+  (js/clearTimeout (:show-0 @timers))
+  (js/clearTimeout (:show-1 @timers))
+  (js/clearTimeout (:show-2 @timers)))
 
 (defn effect
-  [{:keys [flat-list-ref scroll-index-lock?]} {:keys [opacity layout border]} index]
-  (rn/use-effect (fn []
-                   (reagent/next-tick (fn []
-                                        (when @flat-list-ref
-                                          (.scrollToIndex ^js @flat-list-ref
-                                                          #js {:animated false :index index}))))
-                   (js/setTimeout (fn []
-                                    (anim/animate opacity 1)
-                                    (anim/animate layout 0)
-                                    (anim/animate border 12))
-                                  (if platform/ios? 250 100))
-                   (js/setTimeout #(reset! scroll-index-lock? false) 300)
-                   (fn []
-                     (rf/dispatch [:chat.ui/zoom-out-signal nil])
-                     (when platform/android?
-                       (rf/dispatch [:chat.ui/lightbox-scale 1]))))))
+  [{:keys [flat-list-ref scroll-index-lock? timers]} {:keys [opacity layout border]} index]
+  (rn/use-effect
```

Comment:
I think we don't need the mutable approach @OmarBasem, like storing timers in `timers`. The `use-effect` only creates two timers, and we can just store them in let bindings and clear them in the unmount closure.

Also, I think it's a bit better if each use-effect is responsible for cleaning up only its own timers, especially because they're all short-lived.

```clojure
(defn effect
  [{:keys [flat-list-ref scroll-index-lock?]} {:keys [opacity layout border]} index]
  (rn/use-effect
   (fn []
     (reagent/next-tick (fn []
                          (when @flat-list-ref
                            (.scrollToIndex ^js @flat-list-ref
                                            #js {:animated false :index index}))))
     (let [animation-timer (js/setTimeout (fn []
                                            (anim/animate opacity 1)
                                            (anim/animate layout 0)
                                            (anim/animate border 12))
                                          (if platform/ios? 250 100))
           scroll-timer    (js/setTimeout #(reset! scroll-index-lock? false) 300)]
       (fn []
         (rf/dispatch [:chat.ui/zoom-out-signal nil])
         (when platform/android?
           (rf/dispatch [:chat.ui/lightbox-scale 1]))
         (js/clearTimeout animation-timer)
         (js/clearTimeout scroll-timer))))))
```

Or perhaps better without explicit names:

```clojure
(defn effect
  [{:keys [flat-list-ref scroll-index-lock?]} {:keys [opacity layout border]} index]
  (rn/use-effect
   (fn []
     (reagent/next-tick (fn []
                          (when @flat-list-ref
                            (.scrollToIndex ^js @flat-list-ref
                                            #js {:animated false :index index}))))
     (let [timers [(js/setTimeout (fn []
                                    (anim/animate opacity 1)
                                    (anim/animate layout 0)
                                    (anim/animate border 12))
                                  (if platform/ios? 250 100))
                   (js/setTimeout #(reset! scroll-index-lock? false) 300)]]
       (fn []
         (rf/dispatch [:chat.ui/zoom-out-signal nil])
         (when platform/android?
           (rf/dispatch [:chat.ui/lightbox-scale 1]))
         (doseq [t timer] (js/clearTimeout t)))))))
```

Reply:
@ilmotta what about the timers in `toggle-opacity` method? That's not an effect. Do we need to clean them on unmount too?

Reply:
> I think we don't need the mutable approach 

How can I make it work in `toggle-opacity` method without storing the timers in an atom. There is an if condition and 2 branches, and I want to have access to the timers in both branches to clear the timers. Initializing all the timers in a let binding like above I think will execute all the timers at once

Reply:
@OmarBasem, if (and only if) we could eventually have a solution where timers are only created inside `use-effect`, then we wouldn't need to store them in an atom because each `use-effect` would be responsible for cleaning its own timers.

But with the existing code I agree with you, apparently we need the state.